testsuite/cluster_test/src/prometheus.rs

```diff
@@ -0,0 +1,158 @@
+use failure::{
+    self,
+    prelude::{bail, format_err},
+};
+use reqwest::Url;
+use serde::Deserialize;
+use std::{collections::HashMap, env, time::Duration};
+
+pub struct Prometheus {
+    url: Url,
+    client: reqwest::Client,
+}
+
+pub struct MatrixResponse {
+    inner: HashMap<String, TimeSeries>,
+}
+
+pub struct TimeSeries {
+    inner: Vec<(u64, f64)>,
+}
+
+impl Prometheus {
+    pub fn try_new_from_environment() -> Option<Self> {
+        match env::var("PROMETHEUS") {
+            Err(..) => None,
+            Ok(url) => {
+                let url = format!("http://{}:9091", url)
+                    .parse()
+                    .expect("Failed to parse PROMETHEUS");
+                let client = reqwest::Client::new();
+                Some(Self { url, client })
+            }
+        }
+    }
+
+    pub fn query_range(
+        &self,
+        query: String,
+        start: &Duration,
+        end: &Duration,
+        step: u64,
+    ) -> failure::Result<MatrixResponse> {
+        let url = self
+            .url
+            .join(&format!(
+                "api/datasources/proxy/1/api/v1/query_range?query={}&start={}&end={}&step={}",
+                query,
+                start.as_secs(),
+                end.as_secs(),
+                step
+            ))
+            .expect("Failed to make query_range url");
+
+        let mut response = self
+            .client
+            .get(url)
+            .send()
+            .map_err(|e| format_err!("Failed to query prometheus: {:?}", e))?;
+
+        // We don't check HTTP error code here
+        // Prometheus supplies error status in json response along with error text
+
+        let response: PrometheusResponse = response
+            .json()
+            .map_err(|e| format_err!("Failed to parse prometheus response: {:?}", e))?;
+
+        match response.data {
+            Some(data) => MatrixResponse::from_prometheus(data),
+            None => bail!(
+                "Prometheus query failed: {} {}",
+                response.error_type,
+                response.error
+            ),
+        }
+    }
+}
+
+impl MatrixResponse {
+    pub fn avg(&self) -> Option<f64> {
+        if self.inner.is_empty() {
+            return None;
+        }
+        let sum: f64 = self.inner.iter().map(|(_, v)| v.avg()).sum();
```

Comment:
Lol, could you add some comment here? Just a short description of the response format.